File: backend/Api/LeagueSquadApi/Dtos/SquadDtos.cs
namespace LeagueSquadApi.Dtos
{
    public record SquadRequest(string Name, string IconUrl);
    public record SquadResponse(long Id, string Name, string IconUrl, DateTimeOffset CreatedAt);
    public record SquadAggregatedStatsResponse(long Id, long squadId, string Name, string StatName, string StatSquadMember, double StatValue, DateTimeOffset CreatedAt);
}

File: frontend/league-squad-web/src/types/SquadDtos.ts
export interface SquadResponse {
    id: number,
    name: string,
    iconUrl: string,
    createdAt: string
}

// export interface UpdateSquadRequest {
//     name: string,
// }
export type UpdateSquadRequest = Omit<SquadResponse, "id" | "createdAt">;


// export interface CreateSquadRequest {
//     name: string,
// }
export type CreateSquadRequest = Omit<SquadResponse, "id" | "createdAt">;

